genesyscloud/oauth_client/resource_genesyscloud_oauth_client.go

```diff
@@ -80,35 +80,104 @@ func createOAuthClient(ctx context.Context, d *schema.ResourceData, meta interfa
 		return util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to create oauth client %s error: %s", name, err), resp)
 	}
 
-	credentialName := resourcedata.GetNillableValue[string](d, "integration_credential_name")
-	if credentialName != nil {
+	createCredential(ctx, d, client, oauthClientProxy)
 
-		cred_type := "pureCloudOAuthClient"
```

Comment:
Check with @kavinbalagen here.  I put this code here to automatically create an oAuthclient for the accelerator

Reply:
<img width="1152" alt="image" src="https://github.com/user-attachments/assets/cf822d7a-9507-4fd7-98d7-ab3102b7bfe4">

Reply:
it seems to be working John